### ISSUE TITLE
Question: how to use draw render.drawLines()

### ISSUE THREAD

**Issue by thinkrapido** [nim-lang/sdl2#60]
I have a question.

I'm new to sdl2 and nim.

How do I use render.drawLines()?
I have modified the skeleton example according to the documentation.
I don't know how to provide points: ptr Point and what means the count.
Experimenting with pointers I cannot figure out to provide this c vector.

**Comment by def-** [comment]
You have to pass the address to the first point and the number of points that are stored at that location:

```nim
import sdl2

var renderer: RendererPtr

# ...

var points = newSeq[Point]()
points.add((1.cint, 1.cint))
points.add((2.cint, 2.cint))
points.add((3.cint, 3.cint))

renderer.drawLines(addr points[0], points.len.cint)
```

**Comment by thinkrapido** [comment]
wow, cool!

here is what I created to get rid of this addr and len call

```nim
proc drawPolygon(r: RendererPtr, points: var openArray[Point]): SDL_Return {.discardable.} =
  return r.drawLines(addr points[0], points.len.cint)

```

the var at openArray is important

**Comment by def-** [comment]
If you don't want the `var`, you can use `unsafeAddr`:

```nimrod
proc drawPolygon(r: RendererPtr, points: openArray[Point]): SDL_Return {.discardable.} =
  return r.drawLines(unsafeAddr points[0], points.len.cint)
```

**Comment by thinkrapido** [comment]
so proc `addr` and proc unsafeAddr have the same signature.

is it that `addr` is evaluated at compile time during generating the AST and unsafeAddr during runtime?

**Comment by def-** [comment]
No, they should not have the same signature, but do so in 0.13.0. This has been fixed in the current devel branch of Nim already.
